connectors/sources/servicenow.py

```diff
@@ -194,6 +217,83 @@ async def filter_services(self):
             raise
 
+    async def fetch_data(self, url, params):
+        """Hit API call and iterate over the fetched response.
+
+        Args:
+            url (str): ServiceNow query to be executed.
+            params (dict): Params to format the query.
+
+        Yields:
+            dict: Formatted document.
+        """
+
+        try:
+            async for response in self._api_call(url=url, params=params):
+                for record in response:  # pyright: ignore
+                    yield record
+
+        except Exception as exception:
+            logger.warning(
+                f"Skipping data for {url} with {params}. Exception: {exception}."
+            )
+
+    async def fetch_attachment_content(self, metadata, timestamp=None, doit=False):
+        """Fetch attachment content via metadata.
+
+        Args:
+            metadata (dict): Attachment metadata.
+            timestamp (timestamp, None): Attachment last modified timestamp. Defaults to None.
+            doit (bool, False): Whether to get content or not. Defaults to False.
+
+        Returns:
+            dict: Document with id, timestamp & content.
+        """
+
+        attachment_size = int(metadata["size_bytes"])
+        if not (doit and attachment_size > 0):
+            return
+
+        attachment_name = metadata["file_name"]
+        if os.path.splitext(attachment_name)[-1] not in TIKA_SUPPORTED_FILETYPES:
+            logger.warning(f"{attachment_name} is not supported by TIKA, skipping.")
+            return
+
+        if attachment_size > FILE_SIZE_LIMIT:
+            logger.warning(
+                f"File size {attachment_size} of file {attachment_name} is larger than {FILE_SIZE_LIMIT} bytes. Discarding file content."
+            )
+            return
+
+        document = {"_id": metadata["id"], "_timestamp": metadata["_timestamp"]}
+
+        temp_filename = ""
+        async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
+            temp_filename = str(async_buffer.name)
+
+            try:
+                async for response in self._api_call(
+                    url=ENDPOINTS["DOWNLOAD"].format(sys_id=metadata["id"]),
+                    params={},
+                    is_attachment=True,
+                ):
+                    async for data, _ in response.content.iter_chunks():
+                        await async_buffer.write(data)
+
+            except Exception as exception:
+                logger.warning(
+                    f"Skipping content for {attachment_name}. Exception: {exception}."
+                )
+                return
+
+        logger.debug(f"Calling convert_to_b64 for file : {attachment_name}.")
+        await asyncio.to_thread(convert_to_b64, source=temp_filename)
+
+        async with aiofiles.open(file=temp_filename, mode="r") as async_buffer:
+            document["_attachment"] = (await async_buffer.read()).strip()
+        await remove(temp_filename)
```

Comment:
question: Can we encounter an exception here, where the file wasn't deleted properly? If yes we should either retry and/or log a warning, that the file wasn't removed IMO.